private/pkg/git/gittest/gittest.go

```diff
@@ -34,9 +34,48 @@ const (
 	DefaultRemote = "origin"
 )
 
-func ScaffoldGitRepository(t *testing.T) git.Repository {
+type Repository interface {
+	git.Repository
+	Commit(t *testing.T, msg string, files map[string]string, opts ...CommitOption)
+	Checkout(t *testing.T, branch string)
+	CheckoutB(t *testing.T, branch string)
+	Tag(t *testing.T, name string, msg string)
+	Push(t *testing.T)
+	Merge(t *testing.T, branch string)
+	PackRefs(t *testing.T)
+}
+
+type commitOpts struct {
+	executablePaths []string
+}
+
+type CommitOption func(*commitOpts)
+
+func CommitWithExecutableFile(path string) CommitOption {
+	return func(opts *commitOpts) {
+		opts.executablePaths = append(opts.executablePaths, path)
+	}
+}
+
+type scaffoldOptions struct {
+	withOnlyInitialCommit bool
+}
```

Comment:
Same here, I don't think this option is used anywhere now.

Reply:
Done